src/debugger/rnDebugSession.ts

```diff
@@ -127,33 +157,26 @@ export class RNDebugSession extends LoggingDebugSession {
 
                                         if (this.rnCdpProxy) {
                                             this.rnCdpProxy.setApplicationTargetPort(port);
+                                        }
+
+                                        if (this.debugSessionStatus === DebugSessionStatus.ConnectionPending) {
+                                            return;
+                                        }
 
-                                            const attachArguments = {
-                                                type: "pwa-node",
-                                                request: "attach",
-                                                name: "Attach",
-                                                continueOnAttach: true,
-                                                port: this.cdpProxyPort,
-                                                smartStep: false,
-                                            };
-
-                                            vscode.debug.startDebugging(
-                                                this.appLauncher.getWorkspaceFolder(),
-                                                attachArguments,
-                                                this.session
-                                            )
-                                            .then((childDebugSessionStarted: boolean) => {
-                                                if (childDebugSessionStarted) {
-                                                    resolve();
-                                                } else {
-                                                    reject(new Error("Cannot start child debug session"));
-                                                }
-                                            },
-                                            err => {
-                                                reject(err);
-                                            });
-                                        } else {
-                                            throw new Error("Cannot connect to debugger worker: Chrome debugger proxy is offline");
+                                        if (this.debugSessionStatus === DebugSessionStatus.FirstConnection) {
+                                            this.debugSessionStatus = DebugSessionStatus.FirstConnectionPending;
+                                            this.establishDebugSession(resolve);
+                                        } else if (this.debugSessionStatus === DebugSessionStatus.ConnectionAllowed) {
+                                            if (this.nodeSession) {
+                                                this.debugSessionStatus = DebugSessionStatus.ConnectionPending;
+                                                this.reloading = new Promise<void>((resolve) => {
```

Comment:
Do we need that promise? As I can see here is already implemented status checking and when we receive the corresponding signal we can call `establishDebugSession` directly. What do you think?